instrumentation/nginx/src/otel_ngx_module.cpp

```diff
@@ -145,6 +167,74 @@ OtelGetTraceContextVar(ngx_http_request_t* req, ngx_http_variable_value_t* v, ui
   return NGX_OK;
 }
 
+static ngx_int_t
+OtelGetTraceId(ngx_http_request_t* req, ngx_http_variable_value_t* v, uintptr_t data) {
+  TraceContext* traceContext = GetTraceContext(req);
+
+  if (traceContext == nullptr || !traceContext->request_span) {
+    ngx_log_error(
+      NGX_LOG_ERR, req->connection->log, 0,
+      "Unable to get trace context when getting trace id");
+    return NGX_OK;
+  }
+
+  trace::SpanContext spanContext = traceContext->request_span->GetContext();
+
+  if (spanContext.IsValid()) {
+    constexpr int len = 2 * trace::TraceId::kSize;
+    char* data = (char*)ngx_palloc(traceContext->request->pool, len);
```

Comment:
You could also use `req->pool` directly here.
Also `ngx_palloc` might return `NULL`